src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Cells/HealthCertificateQRCodeViewModel.swift

```diff
@@ -15,8 +15,7 @@ struct HealthCertificateQRCodeViewModel {
 		showInfoHit: @escaping () -> Void
 	) {
 		self.base45 = healthCertificate.base45
-		self.shouldBlockCertificateCode = healthCertificate.validityState == .invalid ||
-		(healthCertificate.type != .test && healthCertificate.validityState == .expired)
+		self.shouldBlockCertificateCode = healthCertificate.validityState == .invalid || healthCertificate.validityState == .blocked || (healthCertificate.type != .test && healthCertificate.validityState == .expired)
```

Comment:
Maybe I am wrong, but is this just an other way as `if healthCertificate.validityState == .invalid || healthCertificate.validityState == .blocked || (healthCertificate.type != .test && healthCertificate.validityState != .valid) {` in `HealthCertificateQRCodeCellViewModel.swift` ? 
If so could we move this check to the healthCertifcate itself?

Reply:
Same goes to this:

healthCertificate.validityState == .invalid || healthCertificate.validityState == .blocked || (healthCertificate.type != .test && healthCertificate.validityState != .valid)

It is checked at least twice

Reply:
Very good hind, thanks! There were actually just two different logical chains that I had adjusted all over the place, I tried to consolidate them now, please check the newest commit :)